app/src/main/java/com/sopt/now/ui/login/LoginActivity.kt

```diff
@@ -3,58 +3,119 @@ package com.sopt.now.ui.login
-import com.sopt.now.data.User
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.sopt.now.data.Key.USER_ID
+import com.sopt.now.data.ServicePool
+import com.sopt.now.data.dto.request.RequestLoginDto
+import com.sopt.now.data.dto.response.ResponseLoginDto
 import com.sopt.now.databinding.ActivityLoginBinding
 import com.sopt.now.ui.main.MainActivity
 import com.sopt.now.ui.signup.SignUpActivity
+import org.json.JSONObject
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
 
 class LoginActivity : AppCompatActivity() {
     private lateinit var binding: ActivityLoginBinding
-    private lateinit var user: User
+
+    private val viewModel by viewModels<LoginViewModel>()
+    private var userId: String = ""
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityLoginBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        getUserData()
+
         initSignupBtnClickListener()
+        initView()
+        initObserver()
     }
 
-    private fun getUserData() {
-        val user = intent.getParcelableExtra<User>(USER_PROFILE)
-        if (user != null) {
-            this.user = user
-            initLoginBtnClickListener()
+    private fun initView() {
+        binding.btnLoginToSignin.setOnClickListener {
+            viewModel.login(getLoginRequestDto())
         }
     }
 
+    private fun initObserver() {
+        viewModel.liveData.observe(this) { state ->
+            Toast.makeText(
+                this@LoginActivity,
+                state.message,
+                Toast.LENGTH_SHORT,
+            ).show()
+
+            val intent = Intent(this@LoginActivity, MainActivity::class.java).apply {
+                userId = state.userId ?: ""
+                putExtra(USER_ID, userId)
+            }
+            startActivity(intent)
+        }
+    }
+
+    private fun getLoginRequestDto(): RequestLoginDto {
+        val id = binding.etLoginId.text.toString()
+        val password = binding.etLoginPw.text.toString()
+
+        return RequestLoginDto(
+            authenticationId = id,
+            password = password
+        )
+    }
+
     private fun initSignupBtnClickListener() {
         binding.btnSignupToSignup.setOnClickListener {
             startActivity(Intent(this, SignUpActivity::class.java))
         }
     }
+}
 
-    private fun initLoginBtnClickListener() {
-        binding.btnLoginToSignin.setOnClickListener {
-            checkLogin(user)
-        }
-    }
+class LoginViewModel() : ViewModel() {
+    private val authService by lazy { ServicePool.authService }
 
-    private fun checkLogin(user: User) {
-        if (user.id == binding.etLoginId.text.toString() && user.password == binding.etLoginPw.text.toString()) {
-            val intent = Intent(this, MainActivity::class.java)
+    val liveData = MutableLiveData<LoginState>()
```

Comment:
```suggestion
    val _liveData = MutableLiveData<LoginState>()
    val liveData: MutableLiveData<LoginState> = _liveData
```

제가 언젠가.. 말한 적 있었던 코드 같네요!

Reply:
헝 맞아요! 수정하겠습니다!